formstack-baton-requests/src/main/scala/com/gu/identity/formstackbatonrequests/aws/DynamoClient.scala

```diff
@@ -0,0 +1,65 @@
+package com.gu.identity.formstackbatonrequests.aws
+
+import java.time.format.DateTimeFormatter
+import java.time.LocalDateTime
+
+import com.gu.identity.formstackbatonrequests.SubmissionIdEmail
+import com.typesafe.scalalogging.LazyLogging
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient
+import com.gu.scanamo.Scanamo
+import com.github.t3hnar.bcrypt._
+import com.gu.scanamo.syntax._
+
+trait DynamoClient {
+  def writeSubmissions(submissionIdsAndEmails: List[SubmissionIdEmail], salt: String): List[Either[Throwable, Unit]]
+  def mostRecentTimestamp(): Either[Throwable, SubmissionTableUpdateDate]
+  def updateMostRecentTimestamp(): Either[Throwable, Unit]
+}
+
+case class SubmissionTableUpdateDate(formstackSubmissionTableMetadata: String, date: String)
+
+object SubmissionTableUpdateDate {
+  val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
+}
+
+object Dynamo extends DynamoClient with LazyLogging {
+  private val submissionsTableName = "formstack-submission-ids"
+  private val lastUpdatedTableName = "formstack-submissions-last-updated"
+
+  private val dynamoClient = AmazonDynamoDBClient
+    .builder()
+    .withRegion(AwsCredentials.region)
+    .withCredentials(AwsCredentials.credentials)
+    .build()
+
+  override def writeSubmissions(submissionIdsAndEmails: List[SubmissionIdEmail], salt: String): List[Either[Throwable, Unit]] = {
+    submissionIdsAndEmails.map { submissionIdAndEmail =>
+      for {
+        hashedEmail <- submissionIdAndEmail.email.bcryptSafe(salt).toEither
+        submissionWithHashedEmail = submissionIdAndEmail.copy(email = hashedEmail)
+        _ <- Scanamo.put[SubmissionIdEmail](dynamoClient)(submissionsTableName)(submissionWithHashedEmail)
```

Comment:
i think there should be a way to write things in batch to dynamo. This is writing each record one at a time.

Reply:
Done!